src/nvim/os/env.c

```diff
@@ -243,19 +243,24 @@ void expand_env_esc(char_u *srcp, char_u *dst, int dstlen, bool esc, bool one,
         // Verify that we have found the end of a UNIX ${VAR} style variable
         if (src[1] == '{' && *tail != '}') {
           var = NULL;
-        } else if (src[1] == '{') {
-          ++tail;
-        }
+        } else {
+          if (src[1] == '{') {
+            ++tail;
+          }
 #elif defined(MSWIN)
```

Comment:
Maybe instead of removing the dead code, we should restore the original code. We'll need it eventually, no?